chamge/amqp-edge-backend.c

```diff
@@ -81,51 +103,47 @@ _amqp_declare_queue (const amqp_connection_state_t conn,
         amqp_queue_declare (conn, 1, amqp_cstring_bytes (queue_name), 0, 0, 0,
         1, amqp_empty_table);
   }
-  if (amqp_r == NULL) {
-    g_error ("Can not declare a new queue or queue does not exist...");
-    return CHAMGE_RETURN_FAIL;
-  }
+
+  RETURN_FAIL_IF_TRUE (amqp_r == NULL, "error in amqp_queue_declare");
   amqp_get_rpc_reply (conn);
 
   g_assert_nonnull (amqp_r->queue.bytes);
   response_queue_name = g_strdup (amqp_r->queue.bytes);
-  *queue = amqp_cstring_bytes (response_queue_name);
+  if (queue != NULL)
+    *queue = amqp_cstring_bytes (g_steal_pointer (&response_queue_name));
 
   g_debug ("create private reply_to queue : name -> %s",
       (gchar *) queue->bytes);
   return CHAMGE_RETURN_OK;
 }
 
-static ChamgeReturn
+static void
 _amqp_property_set (amqp_basic_properties_t * const props,
     const guint correlation_id, const gchar * reply_queue_name)
 {
-  gchar *correlation_id_str = NULL;
-  gchar *content_type_str = g_strdup (DEFAULT_CONTENT_TYPE);
+  g_autofree gchar *correlation_id_str = NULL;
+  g_autofree gchar *content_type_str = g_strdup (DEFAULT_CONTENT_TYPE);
 
   props->_flags = AMQP_BASIC_CONTENT_TYPE_FLAG | AMQP_BASIC_DELIVERY_MODE_FLAG;
-  props->content_type = amqp_cstring_bytes (content_type_str);
+  props->content_type =
+      amqp_cstring_bytes (g_steal_pointer (&content_type_str));
   props->delivery_mode = 2;     /* persistent delivery mode */
 
   if (reply_queue_name != NULL) {
-    gchar *reply_to_queue = (gchar *) g_strdup (reply_queue_name);
+    g_autofree gchar *reply_to_queue = (gchar *) g_strdup (reply_queue_name);
     g_assert_nonnull (reply_to_queue);
-    props->reply_to = amqp_cstring_bytes (reply_to_queue);
+    props->reply_to = amqp_cstring_bytes (g_steal_pointer (&reply_to_queue));
```

Comment:
Same as above. it's potentially leaking and can be reduced not to add additional variable.